client/gateway.go

```diff
@@ -743,8 +745,8 @@ func (cli *VanClient) gatewayStartContainer(ctx context.Context, gatewayName str
 		"-d",
 		"--name",
 		gatewayName,
-		"--network",
-		"host",
```

Comment:
I believe this would break `gateway forward` as host ports won't be bound
to the gateway container.

Reply:
It is already broken in non-Linux systems. IIRC, the recommended way to proceed for Linux systems is using podman sites instead. If it is not the case, then we can add a flag to detect the OS running and keep `--network host` for the ones where it is supported.

Reply:
@fgiorgetti Added a check to get the host OS and only do port mapping for `windows` and `darwin`.